listener.py

```diff
@@ -21,16 +21,20 @@ class Listener():
     def __init__(self, thisNode):
         logging.info('Initializing listener')
         self.thisNode = thisNode
+        #Using a UDP socket for catching incoming packets from hosts
         self.sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
```

Comment:
I understand what you are trying to do, but frequent commenting will make the code harder to read, I would suggest avoid commenting too often